src/Microsoft.AspNetCore.SignalR.Common/Internal/Protocol/InvocationMessage.cs

```diff
@@ -58,12 +58,32 @@ public InvocationMessage(string invocationId, bool nonBlocking, string target, E
             Target = target;
             _arguments = arguments;
             _argumentBindingException = argumentBindingException;
+        }
+    }
+
+    public class NonStreamInvocationMessage : InvocationMessage
```

Comment:
I decided to keep it in one file since you are pretty much always interested in the base class (since it has all the functionality) and use derived classes which are basically ctors.